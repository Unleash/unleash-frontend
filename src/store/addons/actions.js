import api from './api';
import { dispatchError } from '../util';

export const RECEIVE_ADDON_CONFIG = 'RECEIVE_ADDON_CONFIG';
export const ERROR_RECEIVE_ADDON_CONFIG = 'ERROR_RECEIVE_ADDON_CONFIG';
export const REMOVE_ADDON_CONFIG = 'REMOVE_ADDON_CONFIG';
export const ERROR_REMOVING_ADDON_CONFIG = 'ERROR_REMOVING_ADDON_CONFIG';
export const ADD_ADDON_CONFIG = 'ADD_ADDON_CONFIG';
export const ERROR_ADD_ADDON_CONFIG = 'ERROR_ADD_ADDON_CONFIG';
export const UPDATE_ADDON_CONFIG = 'UPDATE_ADDON_CONFIG';
export const ERROR_UPDATE_ADDON_CONFIG = 'ERROR_UPDATE_ADDON_CONFIG';

// const receiveAddonConfig = value => ({ type: RECEIVE_ADDON_CONFIG, value });
const addAddonConfig = value => ({ type: ADD_ADDON_CONFIG, value });
const updateAdddonConfig = value => ({ type: UPDATE_ADDON_CONFIG, value });
const removeAddonconfig = value => ({ type: REMOVE_ADDON_CONFIG, value });

const success = (dispatch, type) => value => dispatch({ type, value });

export function fetchAddons() {
    return dispatch =>
        api
            .fetchAll()
            .then(success(dispatch, RECEIVE_ADDON_CONFIG))
            .catch(dispatchError(dispatch, ERROR_RECEIVE_ADDON_CONFIG));
}

export function removeAddon(addon) {
    return dispatch =>
        api
            .remove(addon)
            .then(() => dispatch(removeAddonconfig(addon)))
            .catch(dispatchError(dispatch, ERROR_REMOVING_ADDON_CONFIG));
}

export function createAddon(addon) {
    return dispatch =>
        api
            .create(addon)
            .then(res => res.json())
            .then(value => dispatch(addAddonConfig(value)))
            .catch(dispatchError(dispatch, ERROR_ADD_ADDON_CONFIG));
}

export function updateAddon(addon) {
    return dispatch =>
        api
            .update(addon)
            .then(() => dispatch(updateAdddonConfig(addon)))
            .catch(dispatchError(dispatch, ERROR_UPDATE_ADDON_CONFIG));
}
