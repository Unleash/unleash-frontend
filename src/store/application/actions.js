import api from '../../data/applications-api';

export const RECEIVE_ALL_APPLICATIONS = 'RECEIVE_ALL_APPLICATIONS';
export const ERROR_RECEIVE_ALL_APPLICATIONS = 'ERROR_RECEIVE_ALL_APPLICATIONS';
export const ERROR_UPDATING_APPLICATION_DATA = 'ERROR_UPDATING_APPLICATION_DATA';

export const RECEIVE_APPLICATION = 'RECEIVE_APPLICATION';

const recieveAllApplications = json => ({
    type: RECEIVE_ALL_APPLICATIONS,
    value: json,
});

const recieveApplication = json => ({
    type: RECEIVE_APPLICATION,
    value: json,
});

const errorReceiveApplications = (statusCode, type = ERROR_RECEIVE_ALL_APPLICATIONS) => ({
    type,
    statusCode,
});

export function fetchAll() {
    return dispatch =>
        api
            .fetchAll()
            .then(json => dispatch(recieveAllApplications(json)))
            .catch(error => dispatch(errorReceiveApplications(error)));
}

export function storeApplicationMetaData(appName, key, value) {
    return dispatch =>
        api
            .storeApplicationMetaData(appName, key, value)
            .catch(error => dispatch(errorReceiveApplications(error, ERROR_UPDATING_APPLICATION_DATA)));
}

export function fetchApplication(appName) {
    return dispatch =>
        api
            .fetchApplication(appName)
            .then(json => dispatch(recieveApplication(json)))
            .catch(error => dispatch(errorReceiveApplications(error)));
}
