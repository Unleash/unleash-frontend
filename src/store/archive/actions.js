import api from './api';
import { dispatchAndThrow } from '../util';

export const REVIVE_TOGGLE = 'REVIVE_TOGGLE';
export const RECEIVE_ARCHIVE = 'RECEIVE_ARCHIVE';
export const ERROR_RECEIVE_ARCHIVE = 'ERROR_RECEIVE_ARCHIVE';

const receiveArchive = json => ({
    type: RECEIVE_ARCHIVE,
    value: json.features,
});

const reviveToggle = archiveFeatureToggle => ({
    type: REVIVE_TOGGLE,
    value: archiveFeatureToggle,
});

export function revive(featureToggle) {
    return dispatch =>
        api
            .revive(featureToggle)
            .then(() => dispatch(reviveToggle(featureToggle)))
            .catch(dispatchAndThrow(dispatch, ERROR_RECEIVE_ARCHIVE));
}

export function fetchArchive() {
    return dispatch =>
        api
            .fetchAll()
            .then(json => dispatch(receiveArchive(json)))
            .catch(dispatchAndThrow(dispatch, ERROR_RECEIVE_ARCHIVE));
}
