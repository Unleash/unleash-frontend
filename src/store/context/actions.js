import api from '../../data/context-api';
import { dispatchAndThrow } from '../util';

export const RECEIVE_CONTEXT = 'RECEIVE_CONTEXT';
export const ERROR_RECEIVE_CONTEXT = 'ERROR_RECEIVE_CONTEXT';

export const receiveContext = json => ({
    type: RECEIVE_CONTEXT,
    value: json,
});

export function fetchContext() {
    return dispatch =>
        api
            .fetchContext()
            .then(json => dispatch(receiveContext(json)))
            .catch(dispatchAndThrow(dispatch, ERROR_RECEIVE_CONTEXT));
}
