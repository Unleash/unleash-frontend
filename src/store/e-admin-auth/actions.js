import api from './api';
import { dispatchAndThrow } from '../util';
export const RECIEVE_GOOGLE_CONFIG = 'RECIEVE_GOOGLE_CONFIG';
export const RECIEVE_GOOGLE_CONFIG_ERROR = 'RECIEVE_GOOGLE_CONFIG_ERROR';
export const UPDATE_GOOGLE_AUTH = 'UPDATE_GOOGLE_AUTH';
export const UPDATE_GOOGLE_AUTH_ERROR = 'UPDATE_GOOGLE_AUTH_ERROR';
export const RECIEVE_SAML_CONFIG = 'RECIEVE_SAML_CONFIG';
export const RECIEVE_SAML_CONFIG_ERROR = 'RECIEVE_SAML_CONFIG_ERROR';
export const UPDATE_SAML_AUTH = 'UPDATE_SAML_AUTH';
export const UPDATE_SAML_AUTH_ERROR = 'UPDATE_SAML_AUTH_ERROR';

const debug = require('debug')('unleash:e-admin-auth-actions');

export function getGoogleConfig() {
    debug('Start fetching google-auth config');
    return dispatch =>
        api
            .getGoogleConfig()
            .then(config =>
                dispatch({
                    type: RECIEVE_GOOGLE_CONFIG,
                    config,
                })
            )
            .catch(dispatchAndThrow(dispatch, RECIEVE_GOOGLE_CONFIG_ERROR));
}

export function updateGoogleConfig(data) {
    return dispatch =>
        api
            .updateGoogleConfig(data)
            .then(config => dispatch({ type: UPDATE_GOOGLE_AUTH, config }))
            .catch(dispatchAndThrow(dispatch, UPDATE_GOOGLE_AUTH_ERROR));
}

export function getSamlConfig() {
    debug('Start fetching Saml-auth config');
    return dispatch =>
        api
            .getSamlConfig()
            .then(config =>
                dispatch({
                    type: RECIEVE_SAML_CONFIG,
                    config,
                })
            )
            .catch(dispatchAndThrow(dispatch, RECIEVE_SAML_CONFIG_ERROR));
}

export function updateSamlConfig(data) {
    return dispatch =>
        api
            .updateSamlConfig(data)
            .then(config => dispatch({ type: UPDATE_SAML_AUTH, config }))
            .catch(dispatchAndThrow(dispatch, UPDATE_SAML_AUTH_ERROR));
}
