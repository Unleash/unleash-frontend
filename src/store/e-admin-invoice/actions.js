import api from './api';
import { dispatchError } from '../util';
export const RECEIVE_INVOICES = 'RECEIVE_INVOICES';
export const ERROR_FETCH_INVOICES = 'ERROR_FETCH_INVOICES';

const debug = require('debug')('unleash:e-admin-invoice-actions');

export function fetchInvoices() {
    debug('Start fetching invoices for hosted customer');
    return dispatch =>
        api
            .fetchAll()
            .then(value =>
                dispatch({
                    type: RECEIVE_INVOICES,
                    invoices: value.invoices,
                })
            )
            .catch(dispatchError(dispatch, ERROR_FETCH_INVOICES));
}