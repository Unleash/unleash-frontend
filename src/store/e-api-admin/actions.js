import api from './api';
import { dispatchAndThrow } from '../util';
export const RECIEVE_KEYS = 'RECIEVE_KEYS';
export const ERROR_FETCH_KEYS = 'ERROR_FETCH_KEYS';
export const REMOVE_KEY = 'REMOVE_KEY';
export const REMOVE_KEY_ERROR = 'REMOVE_KEY_ERROR';
export const ADD_KEY = 'ADD_KEY';
export const ADD_KEY_ERROR = 'ADD_KEY_ERROR';

const debug = require('debug')('unleash:e-api-admin-actions');

export function fetchApiKeys() {
    debug('Start fetching api-keys');
    return dispatch =>
        api
            .fetchAll()
            .then(value =>
                dispatch({
                    type: RECIEVE_KEYS,
                    keys: value,
                })
            )
            .catch(dispatchAndThrow(dispatch, ERROR_FETCH_KEYS));
}

export function removeKey(key) {
    return dispatch =>
        api
            .remove(key)
            .then(() => dispatch({ type: REMOVE_KEY, key }))
            .catch(dispatchAndThrow(dispatch, REMOVE_KEY));
}

export function addKey(data) {
    return dispatch =>
        api
            .create(data)
            .then(newToken => dispatch({ type: ADD_KEY, token: newToken }))
            .catch(dispatchAndThrow(dispatch, ADD_KEY_ERROR));
}
