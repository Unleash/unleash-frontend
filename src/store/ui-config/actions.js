import api from './api';
import { dispatchAndThrow } from '../util';

export const RECEIVE_CONFIG = 'RECEIVE_CONFIG';
export const ERROR_RECEIVE_CONFIG = 'ERROR_RECEIVE_CONFIG';

export const receiveConfig = json => ({
    type: RECEIVE_CONFIG,
    value: json,
});

export function fetchUIConfig() {
    return dispatch =>
        api
            .fetchConfig()
            .then(json => dispatch(receiveConfig(json)))
            .catch(dispatchAndThrow(dispatch, ERROR_RECEIVE_CONFIG));
}
