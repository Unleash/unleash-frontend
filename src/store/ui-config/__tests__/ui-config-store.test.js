import reducer from '../index';
import { receiveConfig } from '../actions';

beforeEach(() => {
    localStorage.clear();
});

test('should be default state', () => {
    const state = reducer(undefined, {});
    expect(state.toJS()).toMatchSnapshot();
});

test('should be merged state all', () => {
    const uiConfig = {
        headerBackground: 'red',
        slogan: 'hello',
        environment: 'dev',
    };

    const state = reducer(undefined, receiveConfig(uiConfig));
    expect(state.toJS()).toMatchSnapshot();
});

test('should only update headerBackground', () => {
    const uiConfig = {
        headerBackground: 'black',
    };

    const state = reducer(undefined, receiveConfig(uiConfig));
    expect(state.toJS()).toMatchSnapshot();
});
