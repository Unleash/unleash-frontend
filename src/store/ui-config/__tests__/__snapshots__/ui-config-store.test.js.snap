// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be default state 1`] = `
Object {
  "environment": "",
  "flags": Object {},
  "headerBackground": undefined,
  "links": Array [
    Object {
      "href": "https://unleash.github.io?source=oss",
      "icon": "library_books",
      "title": "User documentation",
      "value": "User documentation",
    },
    Object {
      "href": "https://github.com/Unleash",
      "icon": "c_github",
      "title": "Source code on GitHub",
      "value": "GitHub",
    },
  ],
  "name": "Unleash",
  "slogan": "The enterprise ready feature toggle service.",
  "version": "3.x",
}
`;

exports[`should be merged state all 1`] = `
Object {
  "environment": "dev",
  "flags": Object {},
  "headerBackground": "red",
  "links": Array [
    Object {
      "href": "https://unleash.github.io?source=oss",
      "icon": "library_books",
      "title": "User documentation",
      "value": "User documentation",
    },
    Object {
      "href": "https://github.com/Unleash",
      "icon": "c_github",
      "title": "Source code on GitHub",
      "value": "GitHub",
    },
  ],
  "name": "Unleash",
  "slogan": "hello",
  "version": "3.x",
}
`;

exports[`should only update headerBackground 1`] = `
Object {
  "environment": "",
  "flags": Object {},
  "headerBackground": "black",
  "links": Array [
    Object {
      "href": "https://unleash.github.io?source=oss",
      "icon": "library_books",
      "title": "User documentation",
      "value": "User documentation",
    },
    Object {
      "href": "https://github.com/Unleash",
      "icon": "c_github",
      "title": "Source code on GitHub",
      "value": "GitHub",
    },
  ],
  "name": "Unleash",
  "slogan": "The enterprise ready feature toggle service.",
  "version": "3.x",
}
`;
